### PR TITLE
MySQL partitioning for Zabbix

### PR DESCRIPTION
**Pull request by kelvk**
Zabbix housekeeper can bog down a system significantly when history/trend tables are large, thereby impacting other services that run on the system. Partitioning the tables (http://zabbix.org/wiki/Docs/howto/mysql_partition) and transferring the responsibility of housekeeping the tables to stored procedures yields much better responsiveness.

### CODE REVIEW

**Comment by kelvk** [comment]
This can be cheffed in cleanly.

**Comment by erhudy** [comment]
Testing.

**Comment by kelvk** [comment]
Just pushed a minor change. Ready to test.

**Comment by erhudy** [comment]
Restarting.

File: cookbooks/bcpc/recipes/zabbix-server.rb [hunk 1]
```diff
@@ -230,4 +230,50 @@
...
+            ]
+        end
+    end
+
+    # Partitions need to be pre-extended, so running this as a daily crontab
+    cron 'partition-maintenance-all' do
+        minute '7'
+        hour '0'
+        user 'zabbix'
+        environment ({'MYSQL_PWD' => "#{get_config('mysql-zabbix-password')}"})
```
Comment (erhudy) [inline cookbooks/bcpc/recipes/zabbix-server.rb]: This fails during initial catalog compilation because the password hasn't been generated yet.

Comment (kelvk) [inline cookbooks/bcpc/recipes/zabbix-server.rb]: :(

I rebuilt VMs from snapshot but did not purge the bootstrap. I will fix.

Comment (kelvk) [inline cookbooks/bcpc/recipes/zabbix-server.rb]: Commit updated. I have re-tested by purging the items from data bag and rebuilding 3 monitoring VMs. On a broken build, you can resume by re-running chef as well.

**Comment by erhudy** [comment]
Cheffed in without a problem after the fix. The script called by the cron job ran cleanly (did not delete any data because there was nothing to delete).

**Pull request merged**